app/models/manageiq/providers/azure_stack/cloud_manager/orchestration_stack.rb

```diff
@@ -1,6 +1,58 @@
 class ManageIQ::Providers::AzureStack::CloudManager::OrchestrationStack < ManageIQ::Providers::CloudManager::OrchestrationStack
   require_nested :Status
 
+  def self.raw_create_stack(ems, stack_name, template, options = {})
+    $azure_stack_log.debug("Creating orchestration stack [ems=#{ems.name}, " \
+      "stack_name=#{stack_name}, template=#{template.name}, options=#{options}]")
+    ems.with_provider_connection(:service => :Resources) do |client|
+      # Ensure resource group exists because deployment assumes existing one.
+      client.resource_groups.create_or_update(
+        options[:resource_group],
+        client.model_classes.resource_group.new.tap { |g| g.location = ems.provider_region }
+      )
+      # Deploy into the resource group.
+      deployment = client.model_classes.deployment.new
+      deployment.properties = client.model_classes.deployment_properties.new.tap do |props|
+        props.template = JSON.parse(template.content)
+        props.mode = options[:mode]
+        props.parameters = options[:parameters].transform_values! { |v| { 'value' => v } }
+      end
+
+      client.deployments.create_or_update_async(options[:resource_group], stack_name, deployment)
+      build_ems_ref(ems, options[:resource_group], stack_name)
+    end
+  rescue => err
+    $azure_stack_log.error("stack=[#{stack_name}], error: #{err}")
+    raise MiqException::MiqOrchestrationProvisionError, err.to_s, err.backtrace
+  end
+
+  def raw_delete_stack
+    $azure_stack_log.debug("Deleting orchestration stack (ems=#{ext_management_system.name}, stack_name=#{name})")
+    ext_management_system.with_provider_connection(:service => :Resources) do |client|
+      # TODO(miha-plesko): this only deletes the deployment leaving all resources still there. Need to remove those too.
+      client.deployments.delete(resource_group, name)
+    end
+  rescue => err
+    $azure_stack_log.error("stack=[#{name}], error: #{err}")
+    raise MiqException::MiqOrchestrationDeleteError, err.to_s, err.backtrace
+  end
+
+  def raw_status
+    ext_management_system.with_provider_connection(:service => :Resources) do |client|
+      state = client.deployments.get(resource_group, name).properties.provisioning_state
+      Status.new(state.downcase, state)
```

Comment:
The second argument is the reason. It is very important to retrieve the failure reason from provide. Without clear reason every failure likely produces a support ticket, but often the problem occurs on the provide side rather than the manageiq side.